### ISSUE TITLE
Saving models to relative path fails

### ISSUE THREAD

**Issue by Ax3man** [jlmelville/uwot#131]
When trying to save a UMAP model:
```
save_uwot(orange_UMAP, 'my_folder/orange_UMAP')
```
I get this error:

```
Error in file(tarfile, "wb") : cannot open the connection
In addition: Warning messages:
1: In normalizePath(dirname(filename)) :
  path[1]="my_folder": No such file or directory
2: In file(tarfile, "wb") :
  cannot open file 'my_folder/orange_UMAP': No such file or directory
```

Changing to:

```
save_uwot(orange_umap, uwot:::abspath('my_folder/orange_UMAP'))
```
resolves the issue, suggesting that it is an issue with relative paths.

I think the issue stems from:

https://github.com/jlmelville/uwot/blob/fd5f337bbd24586c05dcf6370004dc61364c0192/R/uwot.R#L3960-L3961

The absolute file path is computed after the change of working directory. Swapping these two lines fixed the issue for me.

**Comment by jlmelville** [comment]
Thank you for the report @Ax3man. And also for specifying the fix. for which I have made you a contributor in the `DESCRIPTION`. This should be fixed in `master`. I don't know when I will next attempt a new submission to CRAN though.

**Comment by Ax3man** [comment]
Much appreciated!

**Comment by jlmelville** [comment]
This should be fixed in the latest (0.2.3) CRAN release.
